File: PULSE.Services/Data/Servicings/ServicingPart.cs
namespace PULSE.Services.Data
{
    public partial class ServicingPart
    {
        public int ServicingId { get; set; }
        public int ProductId { get; set; }
        public decimal? UnitPrice { get; set; }
        public int? Quantity { get; set; }

        public virtual Product Product { get; set; } = null!;
        public virtual Servicing Servicing { get; set; } = null!;
    }
}

File: pulse_admin/lib/data/servicing/servicing_part.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pulse_admin/data/abstract/base_model.dart';
import 'package:pulse_admin/data/part/part.dart';

part 'servicing_part.g.dart';

@JsonSerializable()
class ServicingPart extends BaseModel {
  int? servicingId;
  int? productId;
  double? unitPrice;
  int? quantity;

  Part? product;

  ServicingPart();

  factory ServicingPart.fromJson(Map<String, dynamic> json) =>
      _$ServicingPartFromJson(json);

  Map<String, dynamic> toJson() => _$ServicingPartToJson(this);
}
